### PR TITLE
Use Path::is_dir() in fs::read_dir()'s example.

### PR DESCRIPTION
**Pull request by kennytm**
Basically reverts #25508. The `is_dir()` function has been stable since 1.5.0.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
It's now useless to import `use std::fs::{self` if I'm not mistaken.

Once fixed, it can be merged.

**Comment by kennytm** [comment]
@GuillaumeGomez It is still needed since there is a `fs::read_dir` call on line 1342.

**Comment by GuillaumeGomez** [comment]
Oh right, sorry didn't see that one. Then it's all good. Thanks!

@bors: r+ rollup

File: src/libstd/fs.rs [hunk 1]
```diff
@@ -1338,10 +1338,10 @@ pub fn remove_dir_all<P: AsRef<Path>>(path: P) -> io::Result<()> {
 ///
 /// // one possible implementation of walking a directory only visiting files
 /// fn visit_dirs(dir: &Path, cb: &Fn(&DirEntry)) -> io::Result<()> {
-///     if try!(fs::metadata(dir)).is_dir() {
+///     if dir.is_dir() {
 ///         for entry in try!(fs::read_dir(dir)) {
 ///             let entry = try!(entry);
-///             if try!(fs::metadata(entry.path())).is_dir() {
+///             if try!(entry.file_type()).is_dir() {
```
Comment (alexcrichton) [inline src/libstd/fs.rs]: This is actually changing the meaning of this example because `file_type()` doesn't follow symlinks, perhaps this could change to calling `is_dir` on the return value of `path` to preserve the same behavior?

Comment (GuillaumeGomez) [inline src/libstd/fs.rs]: Arf, I misread the doc then. :-/

My bad.

Comment (kennytm) [inline src/libstd/fs.rs]: @alexcrichton Oops sorry. Since this has been merged, do I submit another PR for this?

Comment (GuillaumeGomez) [inline src/libstd/fs.rs]: @kennytm: Yes please.
